src/components/users/EditProfileForm.js

```diff
@@ -1,12 +1,171 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { useState, useEffect } from 'react';
+import { func } from 'prop-types';
+import { useIntl } from 'react-intl';
+import Loader from 'react-loader-spinner';
+import { useDispatch, useSelector } from 'react-redux';
 
 import { userShape } from '../../constants/shapes';
+import FormInput from '../common/FormInput';
+import { genders, getGenderByValue } from './gendersList';
+import FormSelect from '../common/FormSelect';
+import { genderSelectStyle } from './genderSelectStyle';
+import './edit-profile.scss';
+import { userActions } from '../../actions/user.actions';
+import { userRequest } from '../../actions/user.actions';
+import { userConstants } from '../../constants/user.constants';
 
-const EditProfileForm = ({ user }) => {};
+const EditProfileForm = ({ user, setEditProfile }) => {
+  const intl = useIntl();
+
+  const dispatch = useDispatch();
+
+  const [isSubmitted, setIsSubmitted] = useState(false);
+
+  const { loading, requestError, errorMsg, updated } = useSelector(
+    (state) => state.user
+  );
+
+  const [inputs, setInputs] = useState({
+    name: user.name,
+    gender: user.gender,
+    password: '',
+    confirmNewPassword: '',
+    selectGender: getGenderByValue(user.gender),
+  });
+  const { name, gender, password, confirmNewPassword, selectGender } = inputs;
+
+  const equalPasswords = password === confirmNewPassword;
+
+  useEffect(() => {
+    if (updated) {
+      alert(
+        intl.formatMessage({
+          id: 'editprofile.success.text',
+        })
+      );
+      setEditProfile(null);
+      dispatch({ type: userConstants.USER_UPDATE_SUCCESS_FINISHED });
+    }
+  }, [updated, setEditProfile, intl, dispatch]);
+
+  const handleChange = ({ target }) => {
+    setIsSubmitted(false);
+    const { name, value } = target;
+    setInputs((inputs) => ({ ...inputs, [name]: value }));
+  };
+
+  const handleChangeGender = (selectGender) => {
+    setIsSubmitted(false);
+    setInputs((inputs) => ({
+      ...inputs,
+      gender: selectGender['value'],
+      selectGender: selectGender,
+    }));
+  };
+
+  const changedNameOrGender =
+    (name && name !== user.name) || gender !== user.gender;
+
+  const changedPassword = password && equalPasswords;
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setIsSubmitted(true);
+    if (changedNameOrGender || changedPassword) {
+      dispatch(userRequest());
+      dispatch(
+        userActions.update(
+          name,
+          gender,
+          password,
+          confirmNewPassword,
+          changedNameOrGender
+        )
+      );
+    }
+  };
+
+  return (
+    <form align="center" className="edit-profile-form" onSubmit={handleSubmit}>
+      <FormInput
+        labelClassName="user-form__text"
+        inputClassName="user-form__input"
+        inputLabel={intl.formatMessage({
+          id: 'editprofile.update.name.text',
+        })}
+        inputType="name"
```

Comment:
This should be "text".

Reply:
what do you mean?

Reply:
Isn't it the html input's type? there is no **name** type, since name must be an alphanumeric, I guess the correct type here is **text**.

Reply:
You're right. My bad